index.js

```diff
@@ -37,34 +69,31 @@ class Main {
     });
   }
 
-  async disconnect() {
-    // This call should be refactored to work with async/await
-    this.exporter.disconnect();
-    await this.microServer.close();
+  _updateMetrics() {
+    metrics.currentBlock.set(this.worker.lastConfirmedBlock);
+    metrics.requestsCounter.inc(this.worker.getNewRequestsCount());
+    metrics.requestsResponseTime.observe(new Date() - this.worker.lastRequestStartTime);
+    metrics.lastExportedBlock.set(this.worker.lastExportedBlock);
   }
 
   async workLoop() {
     while (this.shouldWork) {
-      const lastRequestStartTime = new Date();
-      const events = await this.worker.work();
-      this.worker.lastExportTime = Date.now();
-      metrics.currentBlock.set(this.worker.lastConfirmedBlock);
+      this.worker.lastRequestStartTime = new Date();
+      const events = await this.worker.work()
+        .catch((err) => { throw new Error(`Error in workloop: ${err.toString()}`); });
 
-      // This metric is intended to count the requests towards the Node endpoint.
-      // The counting is done inside the worker and we fetch the reult here.
-      metrics.requestsCounter.inc(this.worker.getNewRequestsCount());
-      metrics.requestsResponseTime.observe(new Date() - lastRequestStartTime);
-      metrics.lastExportedBlock.set(this.worker.lastExportedBlock);
+      this.worker.lastExportTime = Date.now();
 
-      // Get the position to store in Zookeeper as constructed by the worker.
-      // Different workers may store different type of position so this is
-      // part of the blockchain specific code.
+      this._updateMetrics();
       this.lastProcessedPosition = this.worker.getLastProcessedPosition();
 
       if (events && events.length > 0) {
-        await storeEvents(this.exporter, events);
+        await this.exporter.storeEvents(events)
+          .catch((err) => { throw new Error(`Error in workloop: ${err.toString()}`); });
```

Comment:
does it make sense to catch and instantly throw exception here? as far as I can see behavior is the same.